### PR TITLE
FIX: Various (`~10`) `href`s - `beta.reactjs.org` -> `react.dev` 🔗

### PR DESCRIPTION
**Pull request by Pranav-yadav**
## Summary

This diff fixes various `href`s - `beta.reactjs.org` -> `react.dev` 🔗 across `react.dev` codebase :)

## How did you test this change?

I opened the old links in the browser; which redirect to the new pages and replaced the with the same :)

>**NOTE**: I have skipped the hrefs from blogs, for example "[/intro-to-hooks.html](reactjs.org/docs/hooks-intro.html)" is not auto redirected, though their is no explicit learn page for hooks; this API ref. https://react.dev/reference/react looks right candidate for auto redirect at least for now.

Finally ran:
- `yarn check-all` --> `#00ff00`

### CODE REVIEW

**Comment by Pranav-yadav** [comment]
>**NOTE**: I have skipped the hrefs from blogs, for example "[/intro-to-hooks.html](reactjs.org/docs/hooks-intro.html)" is not auto redirected, though their is no explicit learn page for hooks; this API ref. https://react.dev/reference/react looks right candidate for auto redirect at least for now.

@gaearon WDYT?

File: src/content/blog/2021/12/17/react-conf-2021-recap.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ Looking further into the future, [Xuan Huang (黄玄)](https://twitter.com/Huxpr
 
 ## React docs keynote {/*react-docs-keynote*/}
 
-[Rachel Nabors](https://twitter.com/rachelnabors) kicked off a section of talks about learning and designing with React with a keynote about our investment in React's [new docs](https://beta.reactjs.org/):
+[Rachel Nabors](https://twitter.com/rachelnabors) kicked off a section of talks about learning and designing with React with a keynote about our investment in React's [new docs](https://react.dev/):
```
Comment (gaearon) [inline src/content/blog/2021/12/17/react-conf-2021-recap.md:98]: I'd change that to "React's new docs ([now shipped as react.dev](https://react.dev/blog/2023/03/16/introducing-react-dev))"

File: src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ Static Site Generation (SSG) and Incremental Static Regeneration (ISR) are great
 
 We gave an [early preview](https://www.youtube.com/watch?v=lGEMwh32soc) of React Forget at React Conf 2021. It’s a compiler that automatically generates the equivalent of `useMemo` and `useCallback` calls to minimize the cost of re-rendering, while retaining React’s programming model.
 
-Recently, we finished a rewrite of the compiler to make it more reliable and capable. This new architecture allows us to analyze and memoize more complex patterns such as the use of [local mutations](https://beta.reactjs.org/learn/keeping-components-pure#local-mutation-your-components-little-secret), and opens up many new compile-time optimization opportunities beyond just being on par with memoization hooks.
+Recently, we finished a rewrite of the compiler to make it more reliable and capable. This new architecture allows us to analyze and memoize more complex patterns such as the use of [local mutations](https://react.dev/learn/keeping-components-pure#local-mutation-your-components-little-secret), and opens up many new compile-time optimization opportunities beyond just being on par with memoization hooks.
```
Comment (gaearon) [inline src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md:44]: This should just use a root URL like `/learn/...`

File: src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md [hunk 3]
```diff
@@ -72,8 +72,8 @@ We are working on a new version for the Interaction Tracing API (tentatively cal
 
 ## New React Docs {/*new-react-docs*/}
 
-Last year, we announced the [beta version](https://beta.reactjs.org/) of the new React documentation website. The new learning materials teach Hooks first and has new diagrams, illustrations, as well as many interactive examples and challenges. We took a break from that work to focus on the React 18 release, but now that React 18 is out, we’re actively working to finish and ship the new documentation.
+Last year, we announced the [beta version](https://react.dev/) of the new React documentation website. The new learning materials teach Hooks first and has new diagrams, illustrations, as well as many interactive examples and challenges. We took a break from that work to focus on the React 18 release, but now that React 18 is out, we’re actively working to finish and ship the new documentation.
```
Comment (gaearon) [inline src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md:75]: Here, too: "we announced the beta version of the new React documentation website ([later shipped as react.dev](https://react.dev/blog/2023/03/16/introducing-react-dev)) ..."

**Review by gaearon — CHANGES_REQUESTED** [review]
I outlined a few changes — let's change those, and then change the rest with the same approach

File: src/content/blog/2021/12/17/react-conf-2021-recap.md (hunk 1 above)
Comment (Pranav-yadav) [inline src/content/blog/2021/12/17/react-conf-2021-recap.md:98]:
```suggestion
[Rachel Nabors](https://twitter.com/rachelnabors) kicked off a section of talks about learning and designing with React with a keynote about our investment in React's [new docs](https://react.dev/blog/2023/03/16/introducing-react-dev):
```

File: src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md (hunk 2 above)
Comment (Pranav-yadav) [reply src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md:44]: Yeah, sure. I totally forgot we use MDX hahaha

File: src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md (hunk 3 above)
Comment (Pranav-yadav) [inline src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md:75]:
```suggestion
Last year, we announced the [beta version](https://react.dev/blog/2023/03/16/introducing-react-dev) of the new React documentation website. The new learning materials teach Hooks first and has new diagrams, illustrations, as well as many interactive examples and challenges. We took a break from that work to focus on the React 18 release, but now that React 18 is out, we’re actively working to finish and ship the new documentation.
```

File: src/content/blog/2021/12/17/react-conf-2021-recap.md (hunk 1 above)
Comment (Pranav-yadav) [reply src/content/blog/2021/12/17/react-conf-2021-recap.md:98]: done.

File: src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md (hunk 3 above)
Comment (Pranav-yadav) [reply src/content/blog/2022/06/15/react-labs-what-we-have-been-working-on-june-2022.md:75]: done.

**Comment by Pranav-yadav** [comment]
ping @gaearon :)
Lmk if anything is missed...

**Pull request merged**